controllers/snapshot/snapshot_adapter.go

```diff
@@ -220,6 +226,9 @@ func (a *Adapter) EnsureCreationOfEphemeralEnvironments() (controller.OperationR
 
 		_, err = a.ensureEphemeralEnvironmentForScenarioExists(&integrationTestScenario, testStatuses, components, allEnvironments)
 		if err != nil {
+			if errors.IsNotFound(err) {
+				return controller.RequeueOnErrorOrStop(fmt.Errorf("stop reconciliation, environment %s doesn't exist in the same namespace as integrationScenario %s, try again after creating environment: %w", integrationTestScenario.Spec.Environment.Name, integrationTestScenario.Name, err))
```

Comment:
How `RequeueOnErrorOrStop` works? I assumed it will retry when error is defined, and we have error always defined here.

Reply:
Yes, you are right, I should use `StopProcessing()` here.

Reply:
Fixed. Thanks.